soroban-env-host/src/test/map.rs

```diff
@@ -338,3 +339,62 @@ fn map_build_bad_element_integrity() -> Result<(), HostError> {
 
     Ok(())
 }
+
+#[test]
+fn initialization_invalid() -> Result<(), HostError> {
+    use crate::EnvBase;
+    let host = observe_host!(Host::default());
+
+    // Out of order keys
+    assert!(host
+        .map_new_from_slices(&["b", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    // Duplicate
+    assert!(host
+        .map_new_from_slices(&["a", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    let buf = vec![0; 6000000];
+    let scv_bytes = ScVal::Bytes(buf.try_into().unwrap());
+    let bytes_val = host.to_host_val(&scv_bytes)?;
+
+    // roughly consumes 6*5=30MiB (> 16 MiB is the limit)
+    let vals = vec![bytes_val; 5];
+    let keys = ["a", "b", "c", "d", "e"];
+
+    let res = host.map_new_from_slices(&keys, &vals.as_slice())?;
+    assert!(host.from_host_val(res.to_val()).is_err());
```

Comment:
@jayz22 this assert is failing if the byte buffer above is 6MB, by works if I change it so 7MB. I'd expect it to pass with 6 because there are 5 of them in the map. Do I not understand the limit here? Also, `map_new_from_slices` doesn't fail because the limit is on the xdr, which isn't what I expected.

Reply:
The 16MiB limit is for xdr conversion, and this one is not doing any xdr conversion. It is just cloning the host object `ScBytes` out into result `ScVal`, so just doing memcpy and memalloc. If you look at the budget, it is failing the memory limit of 40MiB, when you set the buffer size to 7MiB.

Reply:
Ah ok makes sense. Updated.